packages/server/customer-os-api/repository/dashboard_repository.go

```diff
@@ -1516,3 +1517,121 @@ func getCorrectValueType(valueToExtract any) float64 {
+     REDUCE(s = [], sliItem in sliPerContract | [
+         CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 11}) - duration({nanoseconds: 1}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 12}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 10}) - duration({nanoseconds: 1}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 11}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 9}) - duration({nanoseconds: 1}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 10}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 8}) - duration({nanoseconds: 1}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 9}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 7}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 8}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 6}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 7}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 5}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 6}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 4}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 5}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 3}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 4}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 2}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 3}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 1}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 2}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth AND sliItem.sliEndedAt > beginOfMonth + duration({months: 1}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END
+     ]) as sliActivePerContract,
+     REDUCE(s = [], sliItem in sliPerContract | 
+         CASE WHEN sliItem.sliStartedAt < beginOfMonth AND sliItem.sliEndedAt > beginOfMonth + duration({months: 1}) - duration({nanoseconds: 1}) THEN s + sliItem ELSE s END
+     ) as sliActiveInCurrentMonth
+WITH year, month, beginOfMonth, endOfMonth, cid, sliActivePerContract, sliActiveInCurrentMonth, 
+REDUCE(acc = null, i IN RANGE(0, 11) |
```

Comment:
WITH year, month, beginOfMonth, endOfMonth, cid, sliActivePerContract, sliActiveInCurrentMonth,, REDUCE(sumArray = [], i IN RANGE(0, 11) |
    sumArray = sumArray[i] + sliActivePerContract[i]
) AS sumPerMonth

WITH year, month, beginOfMonth, endOfMonth, cid, sliActivePerContract, sliActiveInCurrentMonth,
REDUCE(acc = null, i IN RANGE(0, 11) |
    CASE WHEN sumPerMonth[i] > 0 AND acc IS NULL THEN sumPerMonth[i] ELSE acc END
) AS baseline